olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h

```diff
@@ -284,11 +285,31 @@ class AUTHENTICATION_API AuthenticationClient {
    * application.
    * @return CancellationToken that can be used to cancel the request.
    */
+  OLP_SDK_DEPRECATED("Deprecated. Please use version with scope parameter")
   client::CancellationToken SignInClient(
       const AuthenticationCredentials& credentials,
       const SignInClientCallback& callback,
       const std::chrono::seconds& expires_in = std::chrono::seconds(0));
-
+  /**
+   * @brief Sign in with HERE account client credentials, requests a client
+   * access token that identifies the application or service by providing client
+   * credentials. Client access tokens can not be refreshed, instead request a
+   * new one using client credentials.
+   * @param credentials Client access keys issued for the client by the HERE
+   * Account as part of the onboarding or support process.
+   * @param scope Scope that should be assigned to access token.
+   * @param callback The method to be called when request is completed. In case
+   * of successful client sign-in request, the returned HTTP status is 200.
+   * Otherwise, check the response error.
+   * @param expires_in Optional. The number of seconds until the new access
+   * token expires. Ignored if zero or if greater than default expiration of the
+   * application.
+   * @return CancellationToken that can be used to cancel the request.
+   */
+  client::CancellationToken SignInClient(
+      const AuthenticationCredentials& credentials,
+      boost::optional<std::string> scope, const SignInClientCallback& callback,
+      const std::chrono::seconds& expires_in = std::chrono::seconds(0));
```

Comment:
I think if we do the API change either way we should introduce a new structure: struct SignInProperties which currently contains expires_in and scope.
And then the API will look like:

```cpp
client::CancellationToken SignInClient(const AuthenticationCredentials& credentials, 
                                          const SignInProperties& properties, 
                                          SignInClientCallback&& callback);
```

Reply:
Small revision, pass the callback by copy to align with both dataservice-read and dataservice-write clients.